File: frontend/src/constants/projectCategories.ts
export const PROJECT_CATEGORY_PENDING = 'pending'
export const PROJECT_CATEGORY_PUBLISHED = 'published'
export const PROJECT_CATEGORY_ARCHIVED = 'archived'

export const PROJECT_CATEGORY_VALUES = [
  PROJECT_CATEGORY_PENDING,
  PROJECT_CATEGORY_PUBLISHED,
  PROJECT_CATEGORY_ARCHIVED
] as const

export type ProjectCategory = (typeof PROJECT_CATEGORY_VALUES)[number]

export const DEFAULT_PROJECT_CATEGORY: ProjectCategory = PROJECT_CATEGORY_PENDING

export const PROJECT_CATEGORY_LABELS: Record<ProjectCategory, string> = {
  [PROJECT_CATEGORY_PENDING]: '待调整',
  [PROJECT_CATEGORY_PUBLISHED]: '已完成',
  [PROJECT_CATEGORY_ARCHIVED]: '旧版本'
}

export const PROJECT_CATEGORY_TAG_TYPES: Record<ProjectCategory, '' | 'success' | 'info' | 'warning'> = {
  [PROJECT_CATEGORY_PENDING]: 'warning',
  [PROJECT_CATEGORY_PUBLISHED]: 'success',
  [PROJECT_CATEGORY_ARCHIVED]: 'info'
}

File: utils/project_categories.py
"""Project category constants shared by project management flows."""

from __future__ import annotations

from typing import Any

PROJECT_CATEGORY_PENDING = "pending"
PROJECT_CATEGORY_PUBLISHED = "published"
PROJECT_CATEGORY_ARCHIVED = "archived"

DEFAULT_PROJECT_CATEGORY = PROJECT_CATEGORY_PENDING

PROJECT_CATEGORY_VALUES = (
    PROJECT_CATEGORY_PENDING,
    PROJECT_CATEGORY_PUBLISHED,
    PROJECT_CATEGORY_ARCHIVED,
)


def normalize_project_category(value: Any, fallback: str = DEFAULT_PROJECT_CATEGORY) -> str:
    """Normalize user/project category input to a supported value."""
    if isinstance(value, str):
        candidate = value.strip().lower()
        if candidate in PROJECT_CATEGORY_VALUES:
            return candidate
    return fallback


def is_valid_project_category(value: Any) -> bool:
    """Return True when value can be used as a persisted project category."""
    return isinstance(value, str) and value.strip().lower() in PROJECT_CATEGORY_VALUES
